scripts/runtests.py

```diff
@@ -0,0 +1,128 @@
+    # strip trailing commas from the file
+    valid_json = re.sub(r",(\s*?[\}\]])", r"\1", "\n".join(lines))
+    return valid_json
+
+
+def _get_strict_params(stub_path: str) -> list[str]:
+    with open(_STRICTER_CONFIG_FILE) as file:
+        data = json.loads(_parse_jsonc(file.read()))
+    if stub_path in data["exclude"]:
+        return []
+    return ["-p", _STRICTER_CONFIG_FILE]
+
+
+if __name__ == "__main__":
+    try:
+        path = sys.argv[1]
+    except IndexError:
+        print("Missing path argument in format <folder>/<stub>", file=sys.stderr)
+        sys.exit(1)
+    path_tokens = Path(path).parts
+    assert len(path_tokens) == 2, "Path argument should be in format <folder>/<stub>"
+    folder, stub = path_tokens
+    assert folder in {"stdlib", "stubs"}, "Only the 'stdlib' and 'stubs' folders are supported"
+    stubtest_result: Optional[subprocess.CompletedProcess[bytes]] = None
+    pytype_result: Optional[subprocess.CompletedProcess[bytes]] = None
+
+    # Run formatters first. Order matters.
+    subprocess.run(["python3", "-m", "pycln", path, "--all"])
+    subprocess.run(["python3", "-m", "isort", path])
+    subprocess.run(["python3", "-m", "black", path])
+
+    flake8_result = subprocess.run(["python3", "-m", "flake8", path])
+
+    check_consistent_result = subprocess.run(["python3", "tests/check_consistent.py"])
+    check_new_syntax_result = subprocess.run(["python3", "tests/check_new_syntax.py"])
+
+    pyright_result = subprocess.run(["python3", "tests/pyright_test.py", path] + _get_strict_params(path), stderr=subprocess.PIPE)
+    print(pyright_result.stderr.decode())
+    if b"error running npx" in pyright_result.stderr:
+        pyright_returncode = 0
+        pyright_skipped = True
+    else:
+        pyright_returncode = pyright_result.returncode
+        pyright_skipped = False
+
+    mypy_result = subprocess.run(["python3", "tests/mypy_test.py", path])
+    # If mypy failed, stubtest will fail without any helpful error
+    if mypy_result.returncode == 0:
+        if folder == "stdlib":
+            stubtest_result = subprocess.run(["python3", "tests/stubtest_stdlib.py", stub])
+        else:
+            stubtest_result = subprocess.run(["python3", "tests/stubtest_third_party.py", stub])
+    else:
+        print("Skipping stubtest since mypy failed.")
+
+    if sys.platform == "win32":
+        print("Skipping pytype on Windows. You can run the test with WSL.")
+    else:
+        pytype_result = subprocess.run(["python3", "tests/pytype_test.py", path])
+
+    if folder == "stdlib":
+        regr_test_result = subprocess.run(["python3", "tests/regr_test.py", "stdlib"], stderr=subprocess.PIPE)
+    else:
+        regr_test_result = subprocess.run(["python3", "tests/regr_test.py", stub], stderr=subprocess.PIPE)
+    print(regr_test_result.stderr.decode())
+    # No test means they all ran successfully (0 out of 0). Very few 3rd part stub have regression tests.
```

Comment:
This comment feels at risk of getting out of date :)

```suggestion
    # No test means they all ran successfully (0 out of 0). Not all 3rd-party stubs have regression tests.
```

Reply:
Definitely better.